discovery/discovery_test.go

```diff
@@ -1106,5 +1090,67 @@ func TestEthOrchToDBOrch(t *testing.T) {
 	assert.Equal(dbo.ServiceURI, o.ServiceURI)
 	assert.Equal(dbo.EthereumAddr, o.Address.Hex())
 	assert.Equal(dbo.ActivationRound, o.ActivationRound.Int64())
-	assert.Equal(dbo.DeactivationRound,  int64(math.MaxInt64))
+	assert.Equal(dbo.DeactivationRound, int64(math.MaxInt64))
+}
+
+func TestOrchestratorPool_ShuffleGetOrchestrators(t *testing.T) {
+	assert := assert.New(t)
+
+	addresses := stringsToURIs([]string{"https://127.0.0.1:8936", "https://127.0.0.1:8937", "https://127.0.0.1:8938"})
+	//expected := stringsToURIs([]string{"https://127.0.0.1:8938", "https://127.0.0.1:8937", "https://127.0.0.1:8936"})
+
+	ch := make(chan *url.URL)
+
+	oldOrchInfo := serverGetOrchInfo
+	defer func() { serverGetOrchInfo = oldOrchInfo }()
+	serverGetOrchInfo = func(ctx context.Context, bcast common.Broadcaster, server *url.URL) (*net.OrchestratorInfo, error) {
+		ch <- server
+		return nil, nil
+	}
+
+	pool := NewOrchestratorPool(nil, addresses)
+
+	// Check that randomization happens: check for elements in a different order
+	// Sometimes this could fail due to scheduling randomization
```

Comment:
Correct - more specifically, it's because we don't control the order of execution of the GetOrchestratorInfo calls:

https://github.com/livepeer/go-livepeer/blob/cc7ab9d78680569de817927e250f7c9140163c01/discovery/discovery.go#L83-L85

Updated the language a bit in 09b658c502b09b14f38e6150dd7d2cf35a41cd22 ; hopefully it's clearer now (but happy to make further adjustments)